src/libs/actions/connections/Xero.ts

```diff
@@ -0,0 +1,188 @@
+import isObject from 'lodash/isObject';
+import type {OnyxEntry, OnyxUpdate} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {ConnectPolicyToAccountingIntegrationParams, UpdateXeroGenericTypeParams} from '@libs/API/parameters';
+import {READ_COMMANDS, WRITE_COMMANDS} from '@libs/API/types';
+import {getCommandURL} from '@libs/ApiUtils';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
+import type * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+import type {Connections, XeroTrackingCategory} from '@src/types/onyx/Policy';
+
+const getXeroSetupLink = (policyID: string) => {
+    const params: ConnectPolicyToAccountingIntegrationParams = {policyID};
+    const commandURL = getCommandURL({command: READ_COMMANDS.CONNECT_POLICY_TO_XERO, shouldSkipWebProxy: true});
+    return commandURL + new URLSearchParams(params).toString();
+};
+
+const getTrackingCategories = (policy: OnyxEntry<OnyxTypes.Policy>): Array<XeroTrackingCategory & {value: string}> => {
+    const {trackingCategories} = policy?.connections?.xero?.data ?? {};
+    const {mappings} = policy?.connections?.xero?.config ?? {};
+
+    if (!trackingCategories) {
+        return [];
+    }
+
+    return trackingCategories.map((category) => ({
+        ...category,
+        value: mappings?.[`${CONST.XERO_CONFIG.TRACKING_CATEGORY_PREFIX}${category.id}`] ?? '',
+    }));
+};
+
+function createXeroPendingFields<TSettingName extends keyof Connections['xero']['config']>(
+    settingName: TSettingName,
+    settingValue: Partial<Connections['xero']['config'][TSettingName]>,
+    pendingValue: OnyxCommon.PendingAction,
+) {
+    if (!isObject(settingValue)) {
+        return {[settingName]: pendingValue};
+    }
+
+    return Object.keys(settingValue).reduce<Record<string, OnyxCommon.PendingAction>>((acc, setting) => {
+        acc[setting] = pendingValue;
+        return acc;
+    }, {});
+}
+
+function createXeroErrorFields<TSettingName extends keyof Connections['xero']['config']>(
```

Comment:
NAB - We should probably remove the old versions of these once we're not using them anymore! I'll try to remember that for the other xero pr that's coming up

Reply:
Ah yeah, we can remove it once we replaced all the usages of `updatePolicyXeroConnectionConfig`.